fhirclient/models/fhirsearch.py

```diff
@@ -124,14 +128,35 @@ def perform(self, server):
         bundle = bundle.Bundle(res)
         bundle.origin_server = server
         return bundle
-    
+
+    # Use forward references to avoid circular imports
+    def perform_iter(self, server) -> Iterator['Resource']:
+        """ Perform the search by calling `perform` and return an iterator that yields
+        Bundle instances.
+
+        :param server: The server against which to perform the search
+        :returns: An iterator of Bundle instances
+        """
+
+        first_bundle = self.perform(server)
+        if not first_bundle:
+            return iter([])
+        yield first_bundle
+
     def perform_resources(self, server):
         """ Performs the search by calling `perform`, then extracts all Bundle
         entries and returns a list of Resource instances.
         
         :param server: The server against which to perform the search
         :returns: A list of Resource instances
         """
+        # Old method with deprecation warning
+        warnings.warn(
+            "perform_resources() is deprecated and will be removed in a future release. "
+            "Please use perform_resources_iter() instead.",
+            DeprecationWarning,
+        )
```

Comment:
I'm :+1: on adding the deprecation warning to `perform` (because it encourages dropping stuff on the floor unless you're careful) and `perform_resources` (for performance reasons) -- the warning doesn't need to say why, I'm just expanding on why I think a deprecation is warranted.